File: Simple_Regression/simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sun Mar  4 09:11:19 2018

@author: Varun
"""
#Simple Linear Regression
#matrix of independent variable of X (19,1)
#vector of dependent variable of Y (19, )
# importing library
import numpy as np #mathmatical expression
import matplotlib.pyplot as plt #visualize data 
import pandas as pd  #To read csv by using pandas library

# importing dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
Y = dataset.iloc[:, 1].values

#spliting dataset into training set and test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size = 1/3, random_state = 0)

"""#Feature scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_train)"""

#Fitting Simple linear regression to training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

#Predicting the test set results
y_pred = regressor.predict(X_test)

#Visualising the training set results
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Traing set)')
plt.xlabel('Year of experience')
plt.ylabel('Salary')
plt.show()


plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Traing set)')
plt.xlabel('Year of experience')
plt.ylabel('Salary')
plt.show()

File: Simple_Regression/SimpleLinearRegression.R
# Simple Linear Regression
#importing dataset
dataset = read.csv('Salary_Data.csv')

#splitting dataset into traing set and test set
# install.package(caTools)
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
traing_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#Fitting Simple Linear Regression to training set
#lm - linear model
regressor = lm(formula = Salary ~ YearsExperience, data = traing_set)
#summary(regressor)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)    
#(Intercept)        25592       2646   9.672 1.49e-08 ***  
# * - no statical significance *** - high stastical significance

#  YearsExperience     9365        421  22.245 1.52e-14 ***
#  ---
#  Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

#Predicting the Test set results;
y_pred = predict(regressor, newdata = test_set)

#> y_pred
#2         4         5         8        11        16        20        21 
#37766.77  44322.33  46195.35  55560.43  62115.99  71481.07  81782.66  89274.72 
#24        26 
#102385.84 109877.90 

#Visualize the training result
install.packages('ggplot2')
library(ggplot2)
ggplot() + 
  geom_point(aes(x = traing_set$YearsExperience, y = traing_set$Salary),
             colour = 'red') +
  geom_line(aes(x = traing_set$YearsExperience, y = predict(regressor, newdata = traing_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Training set)') +
  xlab('YearsExperience') +
  ylab('Salary')
  

#Visualize the test result
install.packages('ggplot2')
library(ggplot2)
ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = traing_set$YearsExperience, y = predict(regressor, newdata = traing_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Test set)') +
  xlab('YearsExperience') +
  ylab('Salary')






